compiler/rustc_trait_selection/src/traits/error_reporting/suggestions.rs

```diff
@@ -3727,9 +3721,14 @@ impl<'tcx> TypeErrCtxtExt<'tcx> for TypeErrCtxt<'_, 'tcx> {
                     term: ty_var.into(),
                 },
             )));
+            let body_def_id = self.tcx.hir().local_def_id(body_id);
```

Comment:
```suggestion
            let body_def_id = self.tcx.hir().enclosing_body_owner(body_id);
```
?

Reply:
That sounds good thanks! I think this is the method that I need to call in each of the `local_def_id` usage to have the same  behavior or always